### PR TITLE
chore: Add docstring to all functions with 30+ LOC

### PR DESCRIPTION
**Pull request by aahung**
#### Why is this change necessary?

We often forgot to write docstring for a new function/method, which is very helpful for other people to understand the code.

#### How does it address the issue?

This PR re-enables pylint rule to check functions/methods with 30+ LOC and enforce them to have a docstring. I set it to 30 because there are too many short functions/methods lacking docstring and I couldn't add them all. So let's start with 30. 

#### What side effects does this change have?

No code change. 

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/deploy/deploy_context.py [hunk 1]
```diff
@@ -174,6 +177,42 @@ def deploy(
...
+        tags : List[str]
+            List of tags passed to CloudFormation
+        region : str
+            AWS region to deploy the stack to
+        fail_on_empty_changeset : bool
+            Should fail when changeset is empty
+        confirm_changeset : bool
+            Should wait for customer's confirm before executing the changeset
+
+        Returns
```
Comment (mgrandis) [inline samcli/commands/deploy/deploy_context.py:212]: Not sure this is necessary if the method doesn't return anything (VSCode doesn't add it)

Comment (aahung) [reply samcli/commands/deploy/deploy_context.py:212]: Good question. I don't find standard about this one in the official doc. https://numpydoc.readthedocs.io/en/latest/format.html

In PyCharm, Returns is added even it returns `None`. And this stackoverflow post suggests to leave a None here with 42 upvotes

https://stackoverflow.com/questions/27784179/docstrings-when-nothing-is-returned

> You should use None, as that is what your function actually returns:

Comment (aahung) [reply samcli/commands/deploy/deploy_context.py:212]: update, I think we should remove it, I looked into the numpy source code and numpy itself does not add "Returns None" 

https://github.com/numpy/numpy/blob/5cae51e794d69dd553104099305e9f92db237c53/numpy/lib/format.py#L455-L468

```py
def write_array_header_2_0(fp, d):
    """ Write the header for an array using the 2.0 format.
        The 2.0 format allows storing very large structured arrays.
    .. versionadded:: 1.9.0
    Parameters
    ----------
    fp : filelike object
    d : dict
        This has the appropriate entries for writing its string
        representation to the header of the file.
    """
    _write_array_header(fp, d, (2, 0))
```

File: .pylintrc [hunk 2]
```diff
@@ -74,6 +74,7 @@ disable=
     R0914, # Too many local variables (%s/%s)
     R0915, # Too many statements
     C0415, # Import outside toplevel (%s) Used when an import statement is used anywhere other than the module toplevel. Move this import to the top of the file.
+    C0115, # missing-class-docstring
```
Comment (hoffa) [inline .pylintrc:77]: Why does this have to be disabled?

Comment (aahung) [reply .pylintrc:77]: I won't be able to add docstring to all the classes

**Review by hoffa — APPROVED** [review]
LGTM, thanks! 🙌

File: samcli/commands/deploy/deploy_context.py (hunk 1 above)
Comment (aahung) [reply samcli/commands/deploy/deploy_context.py]: There is actually a pylint rule for this "redundant-returns-doc"

All fixed

**Review by mgrandis — APPROVED** [review]

**Pull request merged**